CHANGELOG.md

```diff
@@ -22,6 +22,7 @@
 - Fixed a bug where errors were only silenced on the main thread.
 - Fixed a memory leak when opening datasets.
 - Avoid creating unaligned references in `H5Type` derive macro.
+- Fixed compression filter not giving an error on non-chunked datasets.
```

Comment:
Now that we've changed the check, maybe reword it as "Applying filters without chunking will now produce an explicit error."?

Reply:
Fixed, thanks for keeping this up to date